backend/src/prereq_loader.ts

```diff
@@ -30,20 +30,28 @@ interface NonEmptyQueryResult {
 // prereq query results can be undefined, if the target class doesn't exist.
 type PrereqQueryResult = undefined | NonEmptyQueryResult;
 
+// the curried loader function, expects a termId before producing a loader.
+type CurriedTermIdLoader = (
+  termId: number
+) => DataLoader<SimpleCourse, PrereqQueryResult>;
+
 /**
  * Asynchronously adds prereqs to a Schedule.
  * Does not do mutation.
  * @param schedules the schedule to add prereqs to
  * @param year the year to grab prereqs from (always uses fall).
  */
 export async function addPrereqsToSchedules(
-  schedules: Schedule[],
-  year: number
+  schedules: Schedule[]
```

Comment:
You got rid of the `year` here. What about the case where the pre-reqs for a course may have changed over time? If this were the case then getting the pre-reqs from the `latestOccurence` would be incorrect for a student who started their program in a previous catalog year. 

I believe selecting the earliest ClassOccurrence, with `termId` >= a student's major `catalogYear`, from the list of all occurrences should take care of this

Reply:
Let me know what your thoughts on this are

Reply:
Thanks Yash!! Prereqs aren't kept from old years though, as some of my friends (catalog 2018) now have the new 2019 prereqs for EECE 2160 (OOD), whereas I didn't when I took it over the summer.